src/services/common/metascan-client.js

```diff
@@ -218,3 +228,57 @@ async function recursiveLookup(dataId, pollingInterval, resolve) {
         resolve(response);
     }
 }
+
+
+/**
+ * https://api.metadefender.com/v4/hash/02637A8DA752611218FCEA1854052B9FF160AFD2/sandbox
+ * 
+ * @param {string} sha1
+ * @returns {Promise}
+ */
+function fileSandboxLookup(sha1) {
+    const restEndpoint = `${MCL.config.metadefenderDomain}/${MCL.config.metadefenderVersion}/hash/${sha1}/sandbox`;
+    // console.log(restEndpoint);
+    const options = {
+        headers: { ...authHeader }
+    };
+
+    return fetch(restEndpoint, options).then(data => data.json());
+}
+
+/**
+ *
+ * @param {string} sha1
+ * @param {number} pollingInterval
+ * @returns {Promise}
+ */
+async function poolForSandboxResults(sha1, pollingInterval) {
+    return new Promise((resolve) => {
+        recursiveSandboxLookup(sha1, pollingInterval, resolve);
+    });
+}
+
+/** 
+ *
+ * @param sha1
+ * @param pollingInterval
+ * @param resolve
+ * @returns {Promise.<void>}
+ */
+async function recursiveSandboxLookup(sha1, pollingInterval, resolve) {
+    const response = await fileSandboxLookup(sha1);
+
+    if (response.error) {
```

Comment:
you should  throw new Error here instead of ignoring it and add a try catch in poolForSandboxResults and Reject the promise with the error